syne_tune/experiments/plotting.py

```diff
@@ -433,6 +434,7 @@ def _aggregrate_results(
         df: pd.DataFrame,
         plot_params: PlotParameters,
         extra_results_keys: Optional[List[str]],
+        dataframe_transform: Optional[DataFrameTransform],
```

Comment:
i think adding the transform here conflates too many functionalities - derivation of new metrics and aggregation in the same function. I would suggest to add a separate function that enriches each results dataset with a new column with the derived metric (e.g. hypervoume). That will effectively remove this functionality from _aggregate and allow it to focus on aggregating a metric (original or newly derived)

Users (or plotting function) would then apply the transformation first and then call the aggregate function

Reply:
On a second note, using a transformation is very flexible but also can be less streamlined + there is a potetial problem of mutating the original results frame, especially if we split new metrics addition from aggregation as suggested.

Maybe it would be enough to have a function that appends a new column to the dataframe based on the original dataframe. Enriching data by appending columns (instead of mutating the whole frame) is a pretty nice pattern since the original data is always available and the new columns are appended one by one which makes it very easy to debug. We can use something with signature 
`new_column_generator: Callable[[pd.DataFrame], pd.Series]`
with the following usage
`dataframe["new column"] = new_column_generator(dataframe)`

Reply:
About your earlier comment: `_aggregate_results` is an internal method, the user calls `plot`. Note that data is loaded and aggregated for each call of `plot`, there is no dataframe which is maintained. So, while I will change it to create a single column, I will not change the fact that is passed to `plot`.